aten/src/ATen/native/Math.h

```diff
@@ -67,6 +68,84 @@ Output was modified to be inf or -inf when input is 1 or -1. */
     POSSIBILITY OF SUCH DAMAGE.
 */
 
+namespace {
+/*
+ * This function is derived from the implementation of the i0e function in the
+ * Cephes Math Library. See note [3-Clause BSD License for the Cephes Math
+ * Library].
+ *
+ * Computes an approximation of the exponentially scaled zeroth order modified
+ * Bessel function of the first kind. The approximation is actually two
+ * (sub)approximations, both using a Chebyshev polynomial expansion. One
+ * approximates the function over [0, 8], and the other over (8, infinity). This
+ * function takes the absolute value of all inputs to convert them into the
+ * domain of the approximation.
+ */
+jiterator_code_stringify(
+    jiterator_code(
+        template <typename T>
+        JITERATOR_HOST_DEVICE T chbevl(T x, const T array[], const int len) {
+          T b0, b1, b2;
+
+          b0 = array[0];
+          b1 = 0;
+
+          for (int i = 1; i < len; ++i) {
+            b2 = b1;
+            b1 = b0;
+            b0 = x * b1 - b2 + array[i];
+          }
+
+          return T{0.5} * (b0 - b2);
+        }
+
+        template <typename T>
+        JITERATOR_HOST_DEVICE T calc_i0e(T _x) {
+          T x = fabs(_x);
+
+          if (x <= T{8.0}) {
+            T coefficients[] = {
```

Comment:
Nit unrelated to this PR. Make this constexp or, at the very least, static (provided cuda allows..)

Reply:
Makes sense. Thanks!